addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1741,6 +1741,61 @@ options.registry.collapse = options.Class.extend({
     },
 });
 
+options.registry.masonry_display = options.Class.extend({
+    xmlDependencies: (options.Class.xmlDependencies || []).concat(
+        ['/website/static/src/snippets/s_masonry_block/s_masonry_block.xml']
+    ),
+
+    /**
+     * @override
+     */
+    start: function (params) {
+        if (!this.$target[0].dataset.currentTemplate) {
+            this.$target[0].dataset.currentTemplate = "default";
+            this.$target.children().first().append($(qweb.render(`website.s_template_masonry.default`, {})));
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Change the masonry snippet layout with templates.
+     *
+     * @see this.selectClass for parameters
+     */
+    editLayout: function (previewMode, widgetValue, params) {
+        this.$target[0].dataset.currentTemplate = widgetValue;
+        const container = this.$target.children().first();
```

Comment:
While you remove the background images and shapes on the masonry snippet in this task, removing the ability to add parallax/shape elements at the top-level of the section, using `this.$target.children().first()` is probably still a little fragile. I suggest you still use an explicit selector for the container here (and in the start method):

```js
const container = this.$('.container, .container-fluid, .o_container_small');
```